### PR TITLE
Add filterOptions.renderValue

### PR DESCRIPTION
**Pull request by Legogris**
Adds option to control display of items in filter controls for checkbox, multiselect and select filter types.

### CODE REVIEW

**Comment by Legogris** [comment]
@gregnb

File: README.md [hunk 1]
```diff
@@ -231,7 +231,7 @@ const columns = [
...
+|**`filterOptions`**|{names, logic, display, renderItem}||(These options affect the filter display and functionality from the filter dialog. To modify the filter chips that display after selecting filters, see `customFilterListOptions`) With filter options, it's possible to use custom names for the filter fields [Example](https://github.com/gregnb/mui-datatables/blob/master/examples/column-filters/index.js), custom filter logic [Example](https://github.com/gregnb/mui-datatables/blob/master/examples/customize-filter/index.js), and custom rendering [Example](https://github.com/gregnb/mui-datatables/blob/master/examples/customize-filter/index.js)
```
Comment (patorjk) [inline README.md:234]: I think this should be renderValue instead of renderItem

File: src/components/TableFilter.js [hunk 2]
```diff
@@ -195,6 +201,9 @@ class TableFilter extends React.Component {
 
   renderTextField(column, index) {
     const { classes, filterList } = this.props;
+    if (column.filterOptions && column.filterOptions.render) {
```
Comment (patorjk) [inline src/components/TableFilter.js:204]: I think this should be renderValue instead of render

**Comment by patorjk** [comment]
Thank you for putting this together and sorry it has taken so long for a review. This appears to be pretty useful and looks good other than the two small comments I made, though I've gone ahead and made the fixes. I've merged this into the v3 branch which will hopefully release later this month.

**Pull request merged**